xfp/xiter.py

```diff
@@ -60,6 +60,26 @@ def __getitem__(self, i: int) -> X:
         """
         return self.get(i)
 
+    def takewhile(self, predicate: Callable[[X], bool]) -> "Xiter[X]":
+        """Return a new iterator that stops yielding elements when predicate = False.
+
+        Do not consume the original iterator.
+
+        Useful to limit an infinite Xiter with a predicate.
+
+        ### Usage
+
+        ```python
+            from xfp import Xiter
+            import itertools
+
+            infinite_xiter = Xiter(itertools.count(start=0,step=2))  # -> Xiter([0,2,4,6,8,...])
+            until_xiter = infinite_xiter.takewhile(lambda x: x<6)    # -> Xiter([0,2,4])
```

Comment:
```suggestion
            (
              Xiter(itertools.count(start=0,step=2))  # -> Xiter([0,2,4,6,8,...])
                .takewhile(lambda x: x<6)    # -> Xiter([0,2,4])
            )
```

renforcer l'usage de dot notation dans les exemples pour assoir les bonnes pratiques ? aussi vérifie mais niveau de l'indentation je me demande si tu as pas un niveau en trop par rapport à la doc existante